### DISCUSSION TITLE
formfile[] example

### DISCUSSION THREAD

**Discussion by guidoDimasi** [elsa-workflows/elsa-core#5394] [Q&A]
can you provide an example workflow that use the uploaded file from a HttpEndpoint?

**Comment by glime-ai** [comment]
Certainly! Below is an example of a workflow that uses the uploaded file from an `HttpEndpoint`. This workflow will:

1. Define an HTTP endpoint that accepts file uploads via a POST request.
2. Read the uploaded file from the request.
3. Perform some processing with the file (this part is abstract and can be replaced with actual logic).

```csharp
using Elsa.Activities.Http;
using Elsa.Builders;

public class ProcessUploadedFileWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .HttpEndpoint(http => http
                .WithPath("/upload")
                .WithMethod(HttpMethods.Post)
                .WithReadContent(true)
                .WithTargetType(typeof(IFormFile))
                .WithName("ReceiveFileEndpoint")
                .CanStartWorkflow())
            .Then(context =>
            {
                // Retrieve the uploaded file from the HTTP request
                var file = (IFormFile)context.GetInput();
                
                // Perform some processing with the file
                // For example, save the file to disk, analyze the content, etc.
                ProcessFile(file);
                
                // Return a response indicating success
                return new HttpResponseModel
                {
                    StatusCode = StatusCodes.Status200OK,
                    Content = "File processed successfully."
                };
            }, name: "ProcessFileActivity");
    }

    private void ProcessFile(IFormFile file)
    {
        // Implement your file processing logic here
        // For example, save the file to a specific directory:
        var filePath = Path.Combine("Uploads", file.FileName);
        using (var stream = new FileStream(filePath, FileMode.Create))
        {
            file.CopyTo(stream);
        }
    }
}
```

In this example, the `HttpEndpoint` activity is configured to handle POST requests at the path `/upload` and read the content as an `IFormFile`. The subsequent anonymous activity represents the processing logic for the uploaded file. You would replace the `ProcessFile` method with your actual file processing code.

Please note that this is a simplified example and does not include error handling or other production considerations such as security checks, file size limits, etc.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by guidoDimasi** [comment]
can you provide the same example but with workflow studio?
